lib/puppet/type/file/content.rb

```diff
@@ -119,6 +120,20 @@ def insync?(is)
       result
     end
 
+    def property_matches?(current, desired)
+      basic = super
+      # The inherited equality is always accepted, so use it if valid.
+      time_types = ["mtime", "ctime"]
+      return basic if basic || !time_types.include?(checksum_type.to_s)
+      return false unless current && desired
+      begin
+        raise "not a time checksum" if !time_types.include?(sumtype(current)) || !time_types.include?(sumtype(desired))
+        DateTime.parse(current) >= DateTime.parse(desired)
```

Comment:
So `current` and `desired` are of the form `{mtime}2015-02-03 23:02:42 -0800`, and I have no idea how `DateTime.parse` is working, but that seems super suprising. It might be better to only parse the `sumdata`: 

```
self.fail "Not a time checksum #{current}" if !time_types.include?(sumtype(current))
self.fail "Not a time checksum #{desired}" if !time_types.include?(sumtype(desired))
begin
  DateTime.parse(sumdata(current)) >= DateTime.parse(sumdata(desired))
rescue => detail
```

Also self.fail (is mixed into each type) and provides context about the resource that generated the errored (both on the console and in the report's logs).